apps/condo/domains/news/schema/ExportNewsRecipientsService.js

```diff
@@ -0,0 +1,205 @@
+
+                let residentsByProperties = []
+                if (!isEmpty(propertyIds)) {
+                    residentsByProperties = await loadListByChunks({
+                        context,
+                        list: Resident,
+                        chunkSize: 50,
+                        where: {
+                            property: {
+                                id_in: propertyIds,
+                            },
+                            deletedAt: null,
+                        },
+                    })
+                }
+
+                const recipientsByOrganization = []
+                if (organizationId && isEmpty(newsItemScopes)) {
+                    const residentsByOrganization = await Resident.getAll(context, {
+                        organization: {
+                            id: organizationId,
+                        },
+                        deletedAt: null,
+                    })
+
+                    const propertiesByOrganization = await loadListByChunks({
+                        context,
+                        list: Property,
+                        chunkSize: 50,
+                        where: {
+                            organization: {
+                                id: organizationId,
+                            },
+                            deletedAt: null,
+                        },
+                    })
+
+                    for (let property of propertiesByOrganization) {
+                        const units = getUnitsFromProperty({ map: property.map })
+
+                        const recipientsData = units.reduce((acc, unit) => {
+                            if (residentsByOrganization.find((resident) => unit.unitName === resident.unitName)) {
+                                acc.push({ address: property.address, unitName: unit.unitName, hasResident: true })
+                                return [...acc]
+                            }
+                            acc.push({ address: property.address, unitName: unit.unitName, hasResident: false })
+                            return [...acc]
+                        }, [])
+                        recipientsByOrganization.push(...recipientsData)
+                    }
+                }
+
+                const recipientsByProperty = []
+                const recipientsByUnitType = []
+                const recipientsByUnitName = []
+                if (!isEmpty(newsItemScopes)) {
+                    for (let newsItemScope of newsItemScopes) {
+                        if (get(newsItemScope, 'property.id')) {
+                            const property = await Property.getOne(context, {
+                                id: newsItemScope.property.id,
+                                deletedAt: null,
+                            })
+
+                            const units = getUnitsFromProperty({ map: property.map })
+
+                            if (!newsItemScope.unitName && !newsItemScope.unitType) {
```

Comment:
You are enumerating all cases of values for unitName and unitType in conditions.
It makes hard to figure out what the code does

AFAIK, for NewsItemScope is logically allowed to set values for fields `unitName` and `unitType` in all combinations, for example, if the house in question does have flat "1" and parking "1", to target news item only to parking "1" we should set `{ unitName: ‘1’,  unitType: ‘parking’ }`. @AleX83Xpert, am I right?

Is it possible to implement filtering in the following way?

```
const unitsFilter = pick(newsItemScope, ['unitName', 'unitType']

const filteredUnits = filter(units, unitsFilter )
const filteredResidents = filter(residentsByProperties, unitsFilter)
```

Reply:
Logically by the way. By specifying `unitName` alone, it is impossible to determine whether this is a parking lot or not. But then you need to redo the template, because the type of room is not indicated there. I made it based on a template because it's the end consumer of the data.